addon/models/program-year.js

```diff
@@ -39,6 +39,50 @@ export default DS.Model.extend(PublishableModel, CategorizableModel, SortableByP
   ),
   requiredPublicationSetFields: ['startYear', 'cohort', 'program'],
   optionalPublicationLengthFields: ['directors', 'competencies', 'terms', 'objectives'],
+
+  /**
+   * All top-level competencies ("domains"), along with their sub-competencies, associated with this program-year.
+   *
+   * @property competencyDomains
+   * @type {Ember.computed}
+   * @public
+   */
+  competencyDomains: computed('competencies.@each.domain', async function() {
+    let promises = [];
+    let domainContainer = {};
+    let domainIds = [];
+    const competencies = await this.get('competencies');
+
+    competencies.forEach(competency => {
+      promises.pushObject(competency.get('domain').then(domain => {
+        if(!domainContainer.hasOwnProperty(domain.get('id'))){
+          domainIds.pushObject(domain.get('id'));
+          domainContainer[domain.get('id')] = ObjectProxy.create({
+            content: domain,
+            subCompetencies: []
+          });
+        }
+        if(competency.get('id') !== domain.get('id')){
+          let subCompetencies = domainContainer[domain.get('id')].get('subCompetencies');
+          if(!subCompetencies.includes(competency)){
+            subCompetencies.pushObject(competency);
+          }
+        }
+      }));
+    });
+
+    await all(promises);
+    return domainIds.map(id => {
+      return domainContainer[id];
+    }).sortBy('title');
+  }),
+
+  /**
+   * @property domains
+   * @public
+   * @type {Ember.computed}
+   * @deprecated Use competencyDomains instead. [ST 2017/08/31]
+   */
   domains: computed('competencies.@each.domain', function(){
     var defer = Ember.RSVP.defer();
```

Comment:
Can you throw an `Ember.deprecate('Use competencyDomains instead.');` message inside of this CP?
Then we get notices when it is called.

Reply:
will do!